lib/improver/tests/weighted_blend/test_PercentileBlendingAggregator.py

```diff
@@ -0,0 +1,270 @@
+
+    def test_3D_simple_case(self):
+        """ Test that for a simple case with only one point and an extra
+            internal dimension behaves as expected"""
+        weights = np.array([0.5, 0.5])
+        percentiles = np.array([0, 50, 100])
+        perc_data = np.array([[[1.0], [2.0]],
+                              [[5.0], [6.0]],
+                              [[10.0], [9.0]]])
+        result = PercentileBlendingAggregator.aggregate(
+            perc_data, 1,
+            percentiles,
+            weights, 0)
+        expected_result = np.array([[1.0], [5.555555], [10.0]])
+        self.assertArrayAlmostEqual(result, expected_result)
+
+    def test_4D_simple_case(self):
+        """ Test that for a simple case with only one point and 4D input data
+            it behaves as expected"""
+        weights = np.array([0.5, 0.5])
+        percentiles = np.array([0, 50, 100])
+        perc_data = np.array([1.0, 3.0, 2.0,
+                              4.0, 5.0, 6.0])
+        input_shape = (3, 2, 1, 1)
+        perc_data = perc_data.reshape(input_shape)
+        result = PercentileBlendingAggregator.aggregate(
+            perc_data, 1,
+            percentiles,
+            weights, 0)
+        expected_result = np.array([[[1.0]], [[3.5]], [[6.0]]])
+        expected_result_shape = (3, 1, 1)
+        self.assertArrayAlmostEqual(result, expected_result)
+        self.assertEqual(result.shape, expected_result_shape)
+
+
+class Test_blend_percentiles(IrisTest):
+    """Test the blend_percentiles method"""
+    def test_blend_percentiles(self):
+        """Test blend_percentile function works"""
+        weights = np.array([0.38872692, 0.33041788, 0.2808552])
+        percentiles = np.array([0., 10., 20., 30., 40., 50.,
+                                60., 70., 80., 90., 100.])
+        result = PercentileBlendingAggregator.blend_percentiles(
+                    PERCENTILE_VALUES, percentiles, weights)
+        expected_result_array = np.array([12.70237152, 16.65161847,
+                                          17.97408712, 18.86356829,
+                                          19.84089805, 20.77406153,
+                                          21.39078426, 21.73778353,
+                                          22.22440125, 23.53863876,
+                                          24.64542338])
+        self.assertArrayAlmostEqual(result, expected_result_array)
+
+    def test_two_percentiles(self):
+        """Test that when two percentiles are provided, the extreme values in
+           the set of thresholds we are blending are returned"""
+        weights = np.array([0.5, 0.5])
+        percentiles = np.array([30., 60.])
+        percentile_values = np.array([[5.0, 8.0], [6.0, 7.0]])
+        result = PercentileBlendingAggregator.blend_percentiles(
+                    percentile_values, percentiles, weights)
+        expected_result = np.array([5.0, 8.0])
+        self.assertArrayAlmostEqual(result, expected_result)
+
```

Comment:
@carolinejjones, raising a technical debt issue would be great. It would be useful if you could include a link to  
[ensemble_copula_coupling.py](https://github.com/metoppv/improver/blob/master/lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py), as I'm not sure there's overlap between this problem, and the problems that have been encountered in ECC, regarding when values are required outside the range provided by the input percentiles.